operator/cmd/root.go

```diff
@@ -113,6 +114,12 @@ var (
 				EnableGatewayAPI: operatorCfg.EnableGatewayAPI,
 			}
 		}),
+
+		cell.Invoke(func(r prometheus.Registerer) {
+			metrics.InitOperatorMetrics()
+			r.MustRegister(metrics.ErrorsWarnings)
```

Comment:
This is leading to a panic when starting the operator hive with `EnableMetrics` == false:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x27abf1c]

goroutine 1 [running]:
github.com/cilium/cilium/operator/cmd.init.func3({0x0, 0x0})
	/home/pippolo/go/src/github.com/cilium/cilium/operator/cmd/root.go:120 +0x1dc
```

Why are we changing the `cell.Invoke` to a `cell.Provide` in `operator/metrics/cell.go`?
We might as well leave that as is and add the above lines in `registerMetricsManager`:

```
    metrics.InitOperatorMetrics()
    r.MustRegister(metrics.ErrorsWarnings)
   metrics.FlushLoggingMetrics()
```

just before appending `mm` to the lifecycle.

Reply:
Yup good point